### ISSUE TITLE
Autoplay Causing The Full Carousel To Delay Load

### ISSUE THREAD

**Issue by benjamin852** [Learus/react-material-ui-carousel#30]
Hi. First great library its working near perfect for me. Here is the carousel I built

```
<Carousel
    animation="slide"
    autoPlay={true}
    interval={9000}
>
     {items.map(test => (
         <Banner item={test} />
      ))}        
</Carousel>

```
My data is loading fine in the banner my problem is I want the interval between pages once the carousel is already loaded to be different than the time it takes for the carousel to load on to the page. 

So When a user opens my page it takes a full 9 seconds for the carousel to load which is a problem since I do want there to be a 9 second interval from the slide effect. Is there anyway I can get the carousel to load instantly but still keep my 9 second interval time? 

Note: Even if I get rid of the interval the carousel still does not load immediately it only loads when I press the button I would like so it no button is needed to press.

Thank you

**Comment by Learus** [comment]
Hello there.

The time the carousel needs to load onto the page doesn't have anything to do with the interval prop.
What that prop does, is, it specifies how much time will pass before the next slide begins its animation (if autoplay is set to true). Not the duration of the animation itself. To customize that, use the "timeout" prop.

I'm trying to test this internally, and I can't seem to be able to reproduce it.
Could you give me more info on your `<Banner/>` implementation? I have reason to believe the issue is elsewhere. Load times could be affected by many things.

**Comment by benjamin852** [comment]
That makes sense so ya I'm a little unsure why there is the delay in loading here is my banner code.


`

    const Banner = props => {
    if (props.newProp) console.log(props.newProp);
    const totalItems = props.length ? props.length : 3;

    let items = [];
    for (let i = 0; i < totalItems; i++) {
        const item = props.item[i];
        console.log(item, 'the item in the banner file');
        const media = (
            <Grid item xs={12 / 3} key={item.token.file.title}>
                <CardMedia
                    style={{ cursor: 'pointer' }}
                    className="Media"
                    image={item.token.file.imageURI}
                    title={item.token.file.title}
                    onClick={() => window.open(item.tokenURI)}
                >
                    <Typography className="MediaCaption">
                        {item.token.file.title}
                    </Typography>
                </CardMedia>
            </Grid>
        );

        items.push(media);
    }

    return (
        <Card raised className="Banner">
            <Grid container spacing={0} className="BannerGrid">
                {items}
            </Grid>
        </Card>
    );
}
export default Banner`


To be a little more clear on what is happening, when I run the app the 3 dots to select which page of the carousel I want to be on do appear right away and if I click on one of the those dots it brings up the carrousel right away so that's good. But the actual data does not load unless I press one of those dots or wait the X seconds which I specify in the interval

**Comment by Learus** [comment]
Firstly, are you sure that the load time is **exactly equal** to the interval prop? How much is the load duration if you set the interval to 0?  Also, what happens if you set the timeout prop to 0?

Lastly, might I suggest using the [Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) function to create your item array?

Example:
```jsx
const Banner = props => {
    return (
        <Card raised className="Banner">
            <Grid container spacing={0} className="BannerGrid">
                {props.item.map( (item) => {
                    console.log(item, 'the item in the banner file');
                    return (
                        <Grid item xs={12 / 3} key={item.token.file.title}>
                            <CardMedia
                                style={{ cursor: 'pointer' }}
                                className="Media"
                                image={item.token.file.imageURI}
                                titl    e={item.token.file.title}
                                onClick={() => window.open(item.tokenURI)}
                            >
                                <Typography className="MediaCaption">
                                    {item.token.file.title}
                                </Typography>
                            </CardMedia>
                        </Grid>
                    )
                }
            </Grid>
        </Card>
    );
}
export default Banner
```

It shouldn't be a problem normally, but it might be relevant given its async nature. Give it a shot and let me know of the results.

**Comment by benjamin852** [comment]
Hi I changed it to map and added the timeout 0 the problem still persists. All the timeout does for me is speed or slow down the speed of the animation. Yes I have a timer and it is equal. I've tried setting the interval to 0 and when I do that it loads automatically but then as you can imagine the banner is not super useful as it is constantly changing.  Thank You

**Comment by benjamin852** [comment]
To anyone who maybe reading this I have not yet found a solution but I have found the source of the problem so I will close this thread now.
The issue is that my items array which I'm mapping over in my carrousel is being populated in an asynchronous useEffect hook that is what's causing a delay in loading the carousel as soon as i refresh the page. I've tested this with a dummy array outside of the hook and it loads fine, I will try find a solution now but that is the problem.
Thank You
